aion/web/controllers/question_controller.ex

```diff
@@ -17,12 +19,16 @@ defmodule Aion.QuestionController do
       question_changeset = Question.changeset(%Question{}, Map.put(question_params, "belongs_to", question_subject))
       case Repo.insert(question_changeset) do
         {:ok, question} ->
-          answer_changeset = Answer.changeset(%Answer{}, %{"content" => answers_content, "belongs_to" => question})
-          case Repo.insert(answer_changeset) do
-            {:ok, answer} ->
-              question
-            {:error, error_answer_changeset} ->
-              Repo.rollback(error_answer_changeset)
+          insert_asnwers =
+            answers_content
+            |> String.split(@answer_separator)
+            |> Enum.map(fn answer_content -> Answer.changeset(%Answer{}, %{"content" => answer_content, "belongs_to" => question}) end)
```

Comment:
Have you thought about extracting these lambdas to actual functions? I think with a proper function name it's much easier to understand what the flow of data in pipeline is (I mean, when you want to get a general idea what's done in this step 'map_to_answer_record' and 'insert_to_repo' seem like a better fit, as at first I only care about **what** the steps are supposed to be doing, later on I can check the functions' implementations' and see **how** they are doing it).